csi/controller_server.go

```diff
@@ -325,29 +312,42 @@ func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi
 	}, nil
 }
 
-func (cs *ControllerServer) waitForVolumeState(volumeID string, state types.VolumeState, notFoundRetry, notFoundReturn bool) bool {
+func isVolumeReady(v *longhornclient.Volume) bool {
+	return v.Ready
+}
+
+func isVolumeAttached(v *longhornclient.Volume) bool {
+	return v.State == string(types.VolumeStateAttached)
+}
+
+func isVolumeDetached(v *longhornclient.Volume) bool {
+	return v.State == string(types.VolumeStateDetached)
+}
+
+func (cs *ControllerServer) waitForVolumeState(volumeID string, wantedState func(v *longhornclient.Volume) bool,
```

Comment:
using `stateFunc` instead of `wantedState`. It's not a state but a function.

Probably rename to `waitForVolumeStateFunc`.